tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -3310,6 +3310,167 @@ llvm::Optional<Value> convertFakeQuantOp(PatternRewriter& rewriter,
+
+    if (pad_before != 0) {
+      // Construct a padding slice for adding values before the contents of
+      // tensor.
+      SmallVector<int64_t> slice_begins;
+      SmallVector<int64_t> slice_shape;
+      for (int i = 0; i < input_type.getRank(); i++) {
+        slice_begins.push_back(slice_before_begin[i]);
+        slice_shape.push_back(slice_before_size[i]);
+      }
+
+      auto slice_before_op = CreateOpAndInfer<tosa::SliceOp>(
+          rewriter, op->getLoc(),
+          RankedTensorType::get(slice_shape, output_type.getElementType()),
+          current_tensor, rewriter.getI64ArrayAttr(slice_begins),
+          rewriter.getI64ArrayAttr(slice_shape));
+
+      // Reverse op is superfluous when the padding value is 1.
+      if (pad_before == 1) {
+        slices.push_back(slice_before_op);
+      } else {
+        auto reverse_before_op = CreateOpAndInfer<tosa::ReverseOp>(
+            rewriter, op->getLoc(), slice_before_op.getType(), slice_before_op,
+            rewriter.getI64IntegerAttr(axis));
+        slices.push_back(reverse_before_op);
+      }
+    }
+
+    // Copy the core tensor
+    slices.push_back(current_tensor);
+
+    if (pad_after != 0) {
+      // Construct a padding slice for adding values after the contents of
+      // tensor.
+      SmallVector<int64_t> slice_begins;
+      SmallVector<int64_t> slice_shape;
+      for (int i = 0; i < input_type.getRank(); i++) {
+        slice_begins.push_back(slice_after_begin[i]);
+        slice_shape.push_back(slice_after_size[i]);
+      }
+
+      auto slice_after_op = CreateOpAndInfer<tosa::SliceOp>(
+          rewriter, op->getLoc(),
+          RankedTensorType::get(slice_shape, output_type.getElementType()),
+          current_tensor, rewriter.getI64ArrayAttr(slice_begins),
+          rewriter.getI64ArrayAttr(slice_shape));
+
+      if (pad_after == 1) {
+        slices.push_back(slice_after_op);
+      } else {
+        auto reverse_after_op = CreateOpAndInfer<tosa::ReverseOp>(
+            rewriter, op->getLoc(), slice_after_op.getType(), slice_after_op,
+            rewriter.getI64IntegerAttr(axis));
+        slices.push_back(reverse_after_op);
+      }
+    }
+
+    // The padded size of each dimension D of the output is:
+    // paddings[D, 0] + tensor.dim_size(D) + paddings[D, 1]
+    int new_padded_size = pad_before + input_type.getDimSize(axis) + pad_after;
+    current_dim_size[axis] = new_padded_size;
+
+    // Create the expected output shape and type.
+    RankedTensorType result_type =
+        RankedTensorType::get(current_dim_size, output_type.getElementType());
+    Value result_value = rewriter.create<tosa::ConstOp>(
+        op->getLoc(), result_type, DenseElementsAttr::get(result_type, {0}));
+
+    // Concatenate the old tensor with padding areas.
+    result = convertConcatV2Op(rewriter, op, result_value, slices, axis);
+
+    if (!result) {
+      (void)rewriter.notifyMatchFailure(op, "fail to create concatV2Op");
```

Comment:
It appears inside convertConcatV2Op notifyMatchFailure is used, if all failure cases there are notified, then here we can jsut return.

Reply:
agree. done.